common/lsp/lsp-protocol-operators.h

```diff
@@ -31,8 +31,12 @@ inline constexpr bool operator<(const Position &a, const Position &b) {
 inline constexpr bool operator>=(const Position &a, const Position &b) {
   return !(a < b);
 }
+inline constexpr bool operator==(const Position &a, const Position &b) {
+  return a.line == b.line && a.character == b.character;
+}
+
 inline constexpr bool rangeOverlap(const Range &a, const Range &b) {
```

Comment:
comment whether abutment counts as overlap

Reply:
Updated comment (abutment does not count as overlap).